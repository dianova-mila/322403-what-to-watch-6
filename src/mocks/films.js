export default [
  {
    name: `Macbeth`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
    backgroundColor: `#F1E9CE`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    rating: 3.3,
    scoresCount: 48798,
    director: `Justin Kurzel`,
    starring: [
      `Michael Fassbender`,
      `Marion Cotillard`,
      `Jack Madigan`
    ],
    runTime: 113,
    genre: `Drama`,
    released: 2015,
    id: `1`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `Beach`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/beach.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/beach.jpg`,
    backgroundColor: `#EBC996`,
    description: `Vicenarian Richard travels to Thailand and finds himself in possession of a strange map. Rumours state that it leads to a solitary beach paradise, a tropical bliss. Excited and intrigued, he sets out to find it.`,
    rating: 3.3,
    scoresCount: 207824,
    director: `Danny Boyle`,
    starring: [
      `Leonardo DiCaprio`,
      `Daniel York`,
      `Patcharawan Patarakijjanon`
    ],
    runTime: 119,
    genre: `Adventure`,
    released: 2000,
    id: `2`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `Dardjeeling Limited`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Dardjeeling_Limited.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/dardjeeling_limited.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Dardjeeling_Limited.jpg`,
    backgroundColor: `#AD9F8B`,
    description: `A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.`,
    rating: 3.6,
    scoresCount: 165106,
    director: `Wes Anderson`,
    starring: [
      `Owen Wilson`,
      `Adrien Brody`,
      `Jason Schwartzman`
    ],
    runTime: 91,
    genre: `Adventure`,
    released: 2007,
    id: `3`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `No Country for Old Men`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/No_Country_for_Old_Men.jpg`,
    backgroundColor: `#BDAD8F`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    rating: 4.1,
    scoresCount: 764976,
    director: `Ethan Coen`,
    starring: [
      `Tommy Lee Jones`,
      `Javier Bardem`,
      `Josh Brolin`
    ],
    runTime: 122,
    genre: `Crime`,
    released: 2007,
    id: `4`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Shutter Island`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Shutter_Island.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/shutter-island.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Shutter_Island.jpg`,
    backgroundColor: `#977461`,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
    rating: 4.1,
    scoresCount: 1002557,
    director: `Martin Scorsese`,
    starring: [
      `Leonardo DiCaprio`,
      `Emily Mortimer`,
      `Mark Ruffalo`
    ],
    runTime: 138,
    genre: `Thriller`,
    released: 2010,
    id: `5`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Matrix`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    backgroundColor: `#B9B27E`,
    description: `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
    rating: 4.4,
    scoresCount: 1503092,
    director: `Wachowski Brothers`,
    starring: [
      `Keanu Reeves`,
      `Laurence Fishburne`,
      `Carrie-Anne Moss`
    ],
    runTime: 136,
    genre: `Action`,
    released: 1999,
    id: `6`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `Orlando`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Orlando.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/orlando.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Orlando.jpg`,
    backgroundColor: `#D8D3BD`,
    description: `Young nobleman Orlando is commanded by Queen Elizabeth I to stay forever young. Miraculously, he does just that. The film follows him as he moves through several centuries of British history, experiencing a variety of lives and relationships along the way, and even changing sex.`,
    rating: 2.6,
    scoresCount: 12292,
    director: `Sally Potter`,
    starring: [
      `Tilda Swinton`,
      `Billy Zane`,
      `Quentin Crisp`
    ],
    runTime: 94,
    genre: `Drama`,
    released: 1992,
    id: `7`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `The Revenant`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Revenant.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/revenant.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Revenant.jpg`,
    backgroundColor: `#92918B`,
    description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    rating: 4,
    scoresCount: 618498,
    director: `Alejandro G. Iñárritu`,
    starring: [
      `Leonardo DiCaprio`,
      `Tom Hardy`,
      `Will Poulter`
    ],
    runTime: 156,
    genre: `Action`,
    released: 2015,
    id: `8`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `Bronson`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    backgroundColor: `#73B39A`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    rating: 3.6,
    scoresCount: 109661,
    director: `Nicolas Winding Refn`,
    starring: [
      `Tom Hardy`,
      `Kelly Adams`,
      `Luing Andrews`
    ],
    runTime: 92,
    genre: `Action`,
    released: 2008,
    id: `9`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `Gangs of new york`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/gangs_of_new_york.jpg`,
    backgroundColor: `#A6B7AC`,
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    rating: 8.8,
    scoresCount: 370881,
    director: `Martin Scorsese`,
    starring: [
      `Leonardo DiCaprio`,
      `Cameron Diaz`,
      `Daniel Day-Lewis`
    ],
    runTime: 167,
    genre: `Crime`,
    released: 2002,
    id: `10`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Moonrise Kingdom`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Moonrise_Kingdom.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/moonrise-kingdom.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Moonrise_Kingdom.jpg`,
    backgroundColor: `#D8E3E5`,
    description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
    rating: 7.9,
    scoresCount: 291183,
    director: `Wes Anderson`,
    starring: [
      `Jared Gilman`,
      `Kara Hayward`,
      `Bruce Willis`
    ],
    runTime: 94,
    genre: `Adventure`,
    released: 2012,
    id: `11`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Johnny English`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Johnny_English.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Johnny_English.jpg`,
    backgroundColor: `#F0DBA2`,
    description: `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
    rating: 10,
    scoresCount: 136843,
    director: `Peter Howitt`,
    starring: [
      `Rowan Atkinson`,
      `John Malkovich`,
      `Natalie Imbruglia`
    ],
    runTime: 88,
    genre: `Comedy`,
    released: 2003,
    id: `12`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `What We Do in the Shadows`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/What-We-Do-in-the-Shadows.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/what-we-do-in-the-shadows.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/What-We-Do-in-the-Shadows.jpg`,
    backgroundColor: `#A39E81`,
    description: `A look into the daily (or rather, nightly) lives of three vampires who've lived together for over 100 years, in Staten Island.`,
    rating: 7.2,
    scoresCount: 6173,
    director: `Jemaine Clement`,
    starring: [
      `Kayvan Novak`,
      `Matt Berry`,
      `Natasia Demetriou`
    ],
    runTime: 30,
    genre: `Comedy`,
    released: 2019,
    id: `13`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `We need to talk about Kevin`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/We_need_to_talk_about_Kevin.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/we-need-to-talk-about-kevin.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/We_need_to_talk_about_Kevin.jpg`,
    backgroundColor: `#E1DFDE`,
    description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
    rating: 3.8,
    scoresCount: 123240,
    director: `Lynne Ramsay`,
    starring: [
      `Tilda Swinton`,
      `John C. Reilly`,
      `Ezra Miller`
    ],
    runTime: 112,
    genre: `Drama`,
    released: 2011,
    id: `14`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Legend`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
    backgroundColor: `#E1DAD7`,
    description: `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
    rating: 3.5,
    scoresCount: 138487,
    director: `Brian Helgeland`,
    starring: [
      `Tom Hardy`,
      `Emily Browning`,
      `Taron Egerton`
    ],
    runTime: 132,
    genre: `Crime`,
    released: 2015,
    id: `15`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `Bohemian Rhapsody`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Bohemian_Rhapsody.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bohemian_rhapsody.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#929FA5`,
    description: `Bohemian Rhapsody is a foot-stomping celebration of Queen, their music and their extraordinary lead singer Freddie Mercury. Freddie defied stereotypes and shattered convention to become one of the most beloved entertainers on the planet. The film traces the meteoric rise of the band through their iconic songs and revolutionary sound. They reach unparalleled success, but in an unexpected turn Freddie, surrounded by darker influences, shuns Queen in pursuit of his solo career. Having suffered greatly without the collaboration of Queen, Freddie manages to reunite with his bandmates just in time for Live Aid. While bravely facing a recent AIDS diagnosis, Freddie leads the band in one of the greatest performances in the history of rock music. Queen cements a legacy that continues to inspire outsiders, dreamers and music lovers to this day.`,
    rating: 6.1,
    scoresCount: 338903,
    director: `Bryan Singer`,
    starring: [
      `Rami Malek`,
      `Lucy Boynton`,
      `Gwilym Lee`
    ],
    runTime: 134,
    genre: `Drama`,
    released: 2018,
    id: `16`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Snatch`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/snatch.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Snatch.jpg`,
    backgroundColor: `#FDFDFC`,
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
    rating: 0.2,
    scoresCount: 716577,
    director: `Guy Ritchie`,
    starring: [
      `Jason Statham`,
      `Brad Pitt`,
      `Benicio Del Toro`
    ],
    runTime: 104,
    genre: `Comedy`,
    released: 2000,
    id: `17`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Seven Years in Tibet`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Seven_Years_in_Tibet.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/seven-years-in-tibet.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Seven_Years_in_Tibet.jpg`,
    backgroundColor: `#C6CADF`,
    description: `True story of Heinrich Harrer, an Austrian mountain climber who became friends with the Dalai Lama at the time of China's takeover of Tibet.`,
    rating: 3.6,
    scoresCount: 112612,
    director: `Jean-Jacques Annaud`,
    starring: [
      `Brad Pitt`,
      `David Thewlis`,
      `BD Wong`
    ],
    runTime: 136,
    genre: `Adventure`,
    released: 1997,
    id: `18`,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    name: `Aviator`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Aviator.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Aviator.jpg`,
    backgroundColor: `#D6CDAF`,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    rating: 9.8,
    scoresCount: 307174,
    director: `Martin Scorsese`,
    starring: [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`
    ],
    runTime: 170,
    genre: `Drama`,
    released: 2014,
    id: `19`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    name: `War of the Worlds`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/War_of_the_Worlds.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/war-of-the-worlds.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/War_of_the_Worlds.jpg`,
    backgroundColor: `#9B7E61`,
    description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
    rating: 9.3,
    scoresCount: 386834,
    director: `Steven Spielberg`,
    starring: [
      `Tom Cruise`,
      `Dakota Fanning`,
      `Tim Robbins`
    ],
    runTime: 116,
    genre: `Adventure`,
    released: 2005,
    id: `20`,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
];
