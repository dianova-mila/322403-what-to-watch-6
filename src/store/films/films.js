import {createReducer} from "@reduxjs/toolkit";
import {changeGenre, loadMovies, getMovieList, loadFavorites} from "../actions";

const initialState = {
  currentGenre: `All genres`,
  movieList: [],
  movies: [],
  isDataLoaded: false,
  favoritesMovies: [],
  isFavoritesLoaded: false
};

const films = createReducer(initialState, (builder) => {
  builder.addCase(changeGenre, (state, action) => {
    state.currentGenre = action.payload;
  });

  builder.addCase(loadMovies, (state, action) => {
    state.movies = action.payload;
    state.movieList = action.payload;
    state.isDataLoaded = true;
  });

  builder.addCase(getMovieList, (state, action) => {
    state.movieList = action.payload;
  });

  builder.addCase(loadFavorites, (state, action) => {
    state.favoritesMovies = action.payload;
    state.isFavoritesLoaded = true;
  });
});

export {films};
