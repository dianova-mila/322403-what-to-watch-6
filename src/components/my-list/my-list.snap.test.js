import React from "react";
import {Router} from "react-router-dom";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import * as redux from "react-redux";
import {createMemoryHistory} from "history";
import MyList from "./my-list";
import films from "../../mocks/films";

const authorizationStatus = true;
const userInfo = {avatarUrl: `img/1.png`};

const onUserAvatarClick = jest.fn();
const onPlayButtonClick = jest.fn();
const onSmallMovieCardClick = jest.fn();
const mockStore = configureStore({});


it(`MyList should render correctly`, () => {
  const fakeDispatch = jest.fn();
  jest.spyOn(redux, `useDispatch`).mockImplementation(() => fakeDispatch);

  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={mockStore({
        USER: {authorizationStatus, userInfo},
        FILMS: {favoritesMovies: films, isFavoritesLoaded: true}
      })}>
        <Router history={history}>
          <MyList
            onUserAvatarClick={onUserAvatarClick}
            onPlayButtonClick={onPlayButtonClick}
            onSmallMovieCardClick={onSmallMovieCardClick}
          />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
