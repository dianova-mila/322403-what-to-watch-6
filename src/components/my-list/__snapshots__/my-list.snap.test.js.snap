// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList should render correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        class="page-title user-page__title"
      >
        My list
      </h1>
      <div
        class="user-block"
      >
        <div
          class="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/1.png"
            width="63"
          />
        </div>
      </div>
    </header>
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Macbeth"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/1/"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Seven Years in Tibet"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/seven-years-in-tibet.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/2/"
            >
              Seven Years in Tibet
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Orlando"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/orlando.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/3/"
            >
              Orlando
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Dardjeeling Limited"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/dardjeeling_limited.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/4/"
            >
              Dardjeeling Limited
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Shutter Island"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/shutter-island.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/5/"
            >
              Shutter Island
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="No Country for Old Men"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/6/"
            >
              No Country for Old Men
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/7/"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <div
            class="small-movie-card__image"
          >
            <img
              alt="Aviator"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg"
              width="280"
            />
          </div>
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/8/"
            >
              Aviator
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
